http-client-core/src/main/java/io/micronaut/http/client/StreamingHttpClientFactory.java

```diff
@@ -47,8 +58,50 @@
      * @param configuration The client configuration
      * @return The client
      * @since 2.2.0
+     * @deprecated Use {@link #createStreamingClient(URI, HttpClientConfiguration)} instead
      */
     @NonNull
-    StreamingHttpClient createStreamingClient(@Nullable URL url, @NonNull HttpClientConfiguration configuration);
+    @Deprecated
+    default StreamingHttpClient createStreamingClient(@Nullable URL url, @NonNull HttpClientConfiguration configuration) {
+        try {
+            return createStreamingClient(url != null ? url.toURI() : null, configuration);
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException(e);
+        }
+    }
+
+    /**
+     * Create a new {@link StreamingHttpClient}. Note that this method should only be used outside of the context of an application. Within Micronaut use
+     * {@link jakarta.inject.Inject} to inject a client instead
+     *
+     * @param uri The base URI
+     * @return The client
+     */
+    @NonNull
+    default StreamingHttpClient createStreamingClient(@Nullable URI uri) {
+        try {
+            return createStreamingClient(uri != null ? uri.toURL() : null);
+        } catch (MalformedURLException e) {
+            throw new UnsupportedOperationException(e);
```

Comment:
better to throw `HttpClientException` or `IllegalArgumentException`

Reply:
I think `UnsupportedOperationException` is better here because this will only happen if `createStreamingClient(URI)` hasn't been implemented yet, and the scheme is ws/wss